packages/react/src/components/Popup/Popup.tsx

```diff
@@ -424,15 +423,15 @@ export default class Popup extends AutoControlledComponent<ReactProps<PopupProps
       }),
     }
 
+    const referenceElement = createPopperReferenceProxy(target || this.triggerRef)
```

Comment:
just to be clear - it seems that there is a reason why this function doesn't accept just HTML element as an arg. I might guess that this is because in ref's case it is necessary to make a delayed evaluation of `current`s bounding rectangle, right?

Just trying to understand what makes guarantees to us that `triggerRef` is assigned with the corresponding DOM element at the moment when Popup is rendered with `open=true` prop?

Reply:
> I might guess that this is because in ref's case it is necessary to make a delayed evaluation of currents bounding rectangle, right?

Yep, `referenceElement` in `react-popper` can be only an HTMLElement or such proxy. We can't pass `triggerRef.current` during render phase because it will be `null` on initial render. But it will be asked by Popper for measures only in effects.